atlascope/core/models/dataset.py

```diff
@@ -43,6 +43,24 @@ def perform_import(self, importer="UploadImporter", **kwargs):
         if not self.name:
             self.name = importer_obj.dataset_name or f'{importer} {self.id}'
 
+    def subimage(self, x0: int, x1: int, y0: int, y1: int) -> 'Dataset':
+        metadata = {
+            'x0': x0,
+            'x1': x1,
+            'y0': y0,
+            'y1': y1,
+        }
+
+        dataset = Dataset(
+            name=f'{self.name} Subimage ({x0}, {y0}) -> ({x1}, {y1})',
+            metadata=metadata,
+            source_dataset=self,
+            content=self.content,
```

Comment:
Are we actually going to crop the content eventually or just read the bounds from the metadata?

Reply:
For now I just wanted to be able to read bounds from metadata--that's the simplest path to be able to work with subimages. As soon as that becomes painful for whatever reason, we can have a method to "reify" these subimages into full-fledged images.